packages/swapper/src/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -118,14 +116,14 @@ export const getThorTradeQuote = async (
 
   switch (tradeType) {
     case TradeType.L1ToL1:
-      return getL1quote(input, streamingInterval, tradeType)
+      return getL1quote(input, deps, streamingInterval, tradeType)
```

Comment:
suggestion: this interface doesn't look very sane and easy to consume, I would expect to have a more common interface such as 

`getL1Quote(input, config)` where config includes dependencies and any other configurations needed to get the l1 quote such as `streamingInterval` and `tradeType`

Same comment for any other `getXQuote` function with dependencies 

I shouldn't block this PR but it might be worth tackling it in a follow up to simplify it from a DX point of view

Reply:
Also agree on this one, but its important we minimise diff in this mega PR so lets do in follow-up